main.js

```diff
@@ -0,0 +1,284 @@
+                     * content begins with a YAML header.
+                     * {Buffer} content: contents of `filename`
+                     */
+
+                    let matter = docmatter(content.toString('UTF-8'));
+
+                    return wrapperTemplate(Object.assign({ // Merge this object with the properties in context.yaml
+                        body: matter.body
+                    }, globalContext, yaml.safeLoad(matter.header)))
+                })
+                .then(html => fse.writeFile(path.join(outDir, filename), html))
+            );
+        return Promise.all(promises)
+    })
+}
+
+/**
+ * 
+ * @param {string} dirname name to find Markdown source files
+ * @param {Handbars.TemplateDelegate} serverTemplate Handlebars template to make the page
+ * @param {Object} globalContext Misc context (navbar, servers, etc)
+ * @returns {Promise} wraps around a Promise.all for each file in `dirname`
+ */
+function servers(dirname, serverTemplate, globalContext) {
+    return fse.readdir(dirname).then(listing => {
+        /**
+         * {String[]} listing: filenames in servers/
+         */
+        let promises = listing.map(filename => 
+                fse.readFile(path.join(dirname,filename))
+                .then(content => {
+                    /**
+                     * {Buffer} content: contents of `filename`
+                     */
+                    let matter = docmatter(content.toString('UTF-8'));
+
+                    return serverTemplate(
+                        Object.assign({
+                            body: Markdown.render(matter.body)
+                        }, globalContext, yaml.safeLoad(matter.header))
+                    );
+                }).then(html => fse.writeFile(path.join(outDir, dirname, filename.replace(/\.md$/i,".html")), html))
+                
+            )
+        return fse.mkdirp(path.join(outDir, dirname)).then(Promise.all(promises));
+    })
+}
+
+/**
+ * Formats a date "Month DD, YYYY"
+ * @param {Date} date
+ * @returns {String} 
+ */
+function formatDate(date) {
+    const monthNames = [
+        "January", "February", "March",
+        "April", "May", "June", "July",
+        "August", "September", "October",
+        "November", "December"
+      ];
+    return `${monthNames[date.getMonth()]} ${date.getDate()}, ${date.getFullYear()}`;
+}
+
+/**
+ * Generates an index of a directories PDF files, and copies them to the HTML folder.
+ * @param {string} dirname Directory to find .PDF files in
+ * @param {string} filename Filename of this page
+ * @param {Handlebars.TemplateDelegate} wrapperTemplate Handlebars template to apply to the whole page
+ * @param {Handlebars.TemplateDelegate} minuteTemplate Template to use on the listing
+ * @param {Object} globalContext Context to use for every page
+ */
+function minutes(dirname, filename, wrapperTemplate, minuteTemplate, globalContext) {
+    return fse.readdir(dirname).then(listing => {
```

Comment:
The `minutes` directory should not be assumed to exist, an error here currently means that the minutes page is not generated at all. Whereas it should be dealt with properly, either by creating a blank minutes page or by creating a blank minutes directory (this leads to the same result page-wise).